qiskit_ibm_runtime/noise_learner/noise_learner.py

```diff
@@ -0,0 +1,257 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Noise learner program."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from dataclasses import asdict, fields
+from typing import Any, Dict, Iterable, Optional, Union
+import logging
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.providers import BackendV2
+from qiskit.primitives.containers import EstimatorPubLike
+from qiskit.primitives.containers.estimator_pub import EstimatorPub
+
+from ..base_primitive import _get_mode_service_backend
+from ..constants import DEFAULT_DECODERS
+from ..runtime_job_v2 import RuntimeJobV2
+from ..ibm_backend import IBMBackend
+from ..options.estimator_options import EstimatorOptions
+from ..options.noise_learner_options import NoiseLearnerOptions
+from ..options.utils import remove_dict_unset_values, remove_empty_dict
+from ..utils import validate_isa_circuits
+from ..utils.utils import is_simulator
+
+from ..fake_provider.local_service import QiskitRuntimeLocalService
+from ..qiskit_runtime_service import QiskitRuntimeService
+
+
+# pylint: disable=unused-import,cyclic-import
+from ..session import Session
+from ..batch import Batch
+
+logger = logging.getLogger(__name__)
+
+
+class NoiseLearner:
+    """Class for executing noise learning experiments.
+
+    The noise learner class allows characterizing the noise processes affecting the gates in one or more
+    circuits of interest, based on the Pauli-Lindblad noise model described in [1].
+
+    The :meth:`run` method allows runnig a noise learner job for a list of circuits. After the job is
+    submitted, the gates are collected into independent layers, and subsequently the resulting layers are
+    are characterized individually. The way in which the gates are collected into layers depends on the
+    ``twirling_strategy`` specified in the given ``options`` (see :class:`NoiseLearnerOptions` for more
+    details).
+
```

Comment:
I think we should spell out a common case that will arise for early adopters, even at the risk of being repetitive. The common case is that they want to force particular full-width (wrt active qubits) layers. (note also that "twirling_strategy" is not an option)

```suggestion
    are characterized individually.
    
    The way in which the gates are collected into layers depends on the twirling
    ``strategy`` specified in the given ``options`` (see :class:`NoiseLearnerOptions` for more
    details).
    Note that all strategies obey barriers.
    For example, if you have three ISA entangling layers of interest, consider putting them into one circuit
    separated by barriers acting on the qubits you wish to twirl, and select ``strategy="active-circuit"``.
```

Reply:
WARNING: the above snippet is actually incorrect because "active-ignore" has a default value ("barrier", "delay"). Can we actually achieve this behaviour without inserting a bunch of dummy gates/measurements? I think "no"